src/v/cloud_storage/async_manifest_view.h

```diff
@@ -120,6 +133,9 @@ class materialized_manifest_cache {
     /// Find manifest by its base offset
     ss::shared_ptr<materialized_manifest> get(model::offset base_offset);
 
+    /// Check manifest by its base offset
+    bool contains(model::offset base_offset);
+
```

Comment:
`materialized_manifest_cache::stop` breaks the semaphore, but i'm not seeing a broken semaphore exception being caught anywhere. Does stop use the exception handling path?